File: RomanToIntegers/RomanToInteger.py
class Solution:
    def romanToInt(self, s):
        syms = ["I", "V", "X", "L", "C", "D", "M"]
        nums = [1, 5, 10,  50, 100, 500, 1000]
        posv = [True, True, True, True, True, True, True]
        
        n = 0
        for i in s[::-1]:
            n += nums[syms.index(i)]
            while posv[syms.index(i)]:
                nums[:syms.index(i)] = [-abs(j) for j in nums[:syms.index(i)]]
                posv[syms.index(i)] = False
        return n
            

File: RomanToIntegers/RomanToInteger.cpp
class Solution {
public:
    int cmp(char ch){
        if(ch == 'M') return 1000;
        if(ch == 'D') return 500;
        if(ch == 'C') return 100;
        if(ch == 'L') return 50;
        if(ch == 'X') return 10;
        if(ch == 'V') return 5;
        return 1;
    }
    int romanToInt(string s) {
        int ans = 0;
        for(int i = 0; i < s.size(); i++){
            ans += cmp(s[i]);
            
            if(cmp(s[i]) < cmp(s[i + 1])){
                ans += cmp(s[i + 1]) - 2 * cmp(s[i]);
                i++;
            }
        }
        return ans;
    }
};